crates/linker/src/macho.rs

```diff
@@ -1286,98 +1295,137 @@ fn surgery_macho_help(
     for sec in rodata_sections
         .iter()
         .chain(bss_sections.iter())
+        // TODO why do we even include uninitialized data if it cannot
```

Comment:
I'm not sure I follow this. Both bss and text sections can have relocations, right? On top of that, we have to copy the data of the these sections to the output executable.

Reply:
I believe it is a hard-error in the linker if a zerofill section carries any relocs. Same is true for ELF as far as I know. I don't think we actually need to copy the data for zerofill - we could instead note down its size, and then prefill with zeroes only the required amount between the end of the last non-zerofill section and the end of the `__DATA` segment (assuming all zerofills are trailing the `__DATA` segment). We need to do this since segments are mmaped from the file according to their file offsets/sizes into vm addresses/sizes, and thus the first N bytes of a zerofill section trailing `__DATA` segment will be mapped as-is from the file, meaning if there's any garbage leftover it will be mapped as-is becoming the initializers for the programs variables.

```
0xaaa __data:
...
0xabc __bss:
00000                                       <- used to init part of the loaded `__bss` section
0xacc __LINKEDIT (or end of __DATA segment)
...
```